src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -835,9 +835,24 @@ trait IterableOps[+A, +CC[X], +C] extends Any {
     *                 corresponding elements of this $coll and `that`. The length
     *                 of the returned collection is the minimum of the lengths of this $coll and `that`.
     */
-  def zip[B](xs: Iterable[B]): CC[(A @uncheckedVariance, B)] = fromIterable(View.Zip(toIterable, xs))
+  def zip[B](xs: Iterable[B]): CC[(A @uncheckedVariance, B)] = zipWith(xs)((_, _))
+
   // sound bcs of VarianceNote
 
+  /** Returns a $coll formed by the result of applying a function to each pair of corresponding elements
+    * from this $coll and another iterable collection. It is semantically equivalent to `(xs zip ys) map f`.
+    * If one of the two collections is longer than the other, its remaining elements are ignored.
+    *
+    *  @param   xs  The iterable providing the second half of each combined pair
+    *  @param   f   The function to apply to each pair of elements
+    *  @tparam  B     the type of the elements in the second half of the combined pairs
+    *  @tparam  R     the type of the elements in the resulting collection
+    *  @return        a new collection of type `That` containing the results of applying the given function `f`
```

Comment:
We should not use `That`, that’s a left over of the standard collections based on `CanBuildFrom` :)

We should just say “a new $Coll containing …”

Reply:
Could you also fix `zip`’s scaladoc at the same time?

Reply:
that's good to know. I was in doubt about it. I guess you meant `$Coll` with lowercase `c`, right?

Reply:
Yeah, `$coll` with lowercase `c` is better, you’re right.